Dockerfile

```diff
@@ -27,11 +27,57 @@ RUN mkdir -p rootfs/"${CRYPTOLPATH}" \
 USER root
 RUN chown -R root:root /cryptol/rootfs
 
+
+from debian:stretch AS solvers
+
+# Install needed packages for building
+RUN apt-get update \
+    && apt-get install -y curl cmake gcc g++ git libreadline-dev
+RUN useradd -m user
+RUN install -d -o user -g user /solvers
+USER user
+WORKDIR /solvers
+RUN mkdir -p rootfs/usr/local/bin
+
+# Build abc from Galois' fork
+RUN git clone https://github.com/GaloisInc/abc.git
```

Comment:
The issue with building from Alan's repo is that he doesn't do releases. In your opinion, should we pick a particular known-good commit? We could also roll the dice and build from HEAD, risking that on any given day there might be breaking changes.

If we choose the latter, we could also do "bleeding-edge" builds of the other solvers, but I thought you might want stable versions for this particular Dockerfile. (I have a bleeding-edge Dockerfile that I like to use personally that I could share if you know of a good home.)

Reply:
The commit the `GaloisInc` repo was based off of was chosen pretty arbitrarily, but you make a good point that it doesn't change frequently. Maybe we could leave it and use that as a way of forcing us to update it intentionally rather than every time Alan commits something.

Reply:
Alan fixed something that was causing it to fault on some Cryptol properties. Before this docker, it would be good to fast forward to master as of today.